src/highrl/envs/teacher_env.py

```diff
@@ -175,23 +177,33 @@ def step(self, action: List) -> Tuple:
         """Step into the new state using an action given by the teacher model
 
         Args:
-            action (List): action to take
+            action (List): action to take (decided difficulty by the teacher model)
+                            which will be used by the env generator to create the robot environment
+                            for the coming robot training session
 
         Returns:
             Tuple: observation, reward, done, info
         """
         self.opt.time_steps += 1
         self.opt.reward = 0.0
-
+        action = action * self.infinite_difficulty
+        ########### Generate Robot Env based on decided difficulty ##############
+        generator_action, _ = self.env_generator.predict(action)
```

Comment:
`generator_action` does not represent the value of the variable, change to `env_points` for example.